task2.js

```diff
@@ -0,0 +1,29 @@
+const { performance } = require("perf_hooks");
+
+const EventEmitter = require("./task1");
+
+class WithTime extends EventEmitter {
+  execute(asyncFunc, ...args) {
+    const functionName = asyncFunc.name;
+    const dataReceived = () => this.emit("data");
+
+    this.emit("begin");
+    performance.mark(`${functionName}-start`);
+    asyncFunc(...args, dataReceived).finally(() => {
```

Comment:
I would prefer explicit `.then``.catch` methods as well. Probably we want to do something with data or error. At least log them.